modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/finder/AMImageFinderImpl.java

```diff
@@ -107,23 +110,29 @@ public Stream<AdaptiveMedia<AMImageProcessor>> getAdaptiveMediaStream(
 		Predicate<AMImageConfigurationEntry> filter =
 			amImageQueryBuilderImpl.getConfigurationEntryFilter();
 
+		List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList =
+			new ArrayList<>();
+
+		for (AMImageConfigurationEntry amImageConfigurationEntry :
```

Comment:
@slnn I think you can also use `ListUtil.filter` on this one.

Reply:
Hi @dantewang 

I did not use` ListUtil.filter` on this one because I do not want to bring additional times of iterations. This code did this: iterates `amImageConfigurationEntries` which is a `Collection<AMImageConfigurationEntry>`,  the item will be used to create an `AdaptiveMedia<AMImageProcessor>` and then added to the list  (`List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList) if it satisfies the condition.


```
`List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList =
			new ArrayList<>();

		for (AMImageConfigurationEntry amImageConfigurationEntry :
				amImageConfigurationEntries) {

			if (filter.test(amImageConfigurationEntry) &&
				_hasAdaptiveMedia(fileVersion, amImageConfigurationEntry)) {

				adaptiveMediaList.add(
					_createMedia(
						fileVersion, uriFactory, amImageConfigurationEntry));
			}
		}`
```

If we use the`  ListUtil.filter`,  the code may look like this: find out all these items that satisfy the condition from 
amImageConfigurationEntries, then iterate it again to create  `AdaptiveMedia<AMImageProcessor>` and then add to the list (`List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList). I do not find any existing API in ListUtil can do it with one iteration.


```
Predicate<AMImageConfigurationEntry> predicate =
			new Predicate<AMImageConfigurationEntry>() {

				@Override
				public boolean test(
					AMImageConfigurationEntry amImageConfigurationEntry) {

					if (filter.test(amImageConfigurationEntry) &&
						_hasAdaptiveMedia(
							fileVersion, amImageConfigurationEntry)) {

						return true;
					}

					return false;
				}

			};

		amImageConfigurationEntries = ListUtil.filter(
			amImageConfigurationEntries, predicate);

		List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList =
			new ArrayList<>();

		for (AMImageConfigurationEntry amImageConfigurationEntry :
				amImageConfigurationEntries) {

			adaptiveMediaList.add(
				_createMedia(
					fileVersion, uriFactory, amImageConfigurationEntry));
		}
```